### PR TITLE
ADEN-4534 add replay functionality

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/ADEN-4534

### CODE REVIEW

File: extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.video.player.ui.replayOverlay',
+		replayOverlayClass = 'replay-overlay';
+
+	function add(video) {
+		var overlay = doc.createElement('div');
+
+		overlay.classList.add(replayOverlayClass);
+		overlay.addEventListener('click', function () {
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js:20]: Shall we store it in Kikimora as a new event for VUAP? //cc @fraszczakszymon @nefretete

File: extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			video.play();
+			log(['replay', log.levels.debug, logGroup]);
+			// make overlay invisible
+			overlay.style.width = '';
+		});
+
+		video.addEventListener('wikiaAdCompleted', function() {
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js:27]: Is it needed mostly for window resizing?

Comment (dianafa) [inline extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js]: yes

Comment (dianafa) [reply extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js:27]: we are un-hiding overlay by assigning it a width when the video ends. When it starts, we want to hide it again

File: skins/oasis/css/core/ads.scss [hunk 3]
```diff
@@ -177,6 +177,15 @@ $uap-video-animation-duration: 0.4s;
 	}
 }
 
+.wikia-ad .replay-overlay {
+	background: rgba(0,0,0,0.7) url('/extensions/wikia/AdEngine/js/video/player/ui/images/replay.svg') center/8% no-repeat;
+	cursor: pointer;
+	height: 100%;
+	position: absolute;
+	top: 0;
+	z-index: 10;
```
Comment (nandy-andy) [inline skins/oasis/css/core/ads.scss]: Where does the magic z-index value come from?

File: extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js (hunk 2 above)
Comment (nandy-andy) [reply extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js:27]: I see. As far as I understand the width of the video is not changing unless somebody resizes the window. I just wonder if we need these width calculations done in `getOverlayWidth()`. We could set it to 100% but then we'll need to set position of the new overlay element to `relative`, I guess.

**Review by nandy-andy — APPROVED** [review]

File: extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	/**
+	 * Basing on width set on video container and total ad width compute width (in %)
+	 * of overlay to make it responsive.
+	 *
+	 * offsetWidth won't work in case video container is hidden.
+	 * @param video
+	 * @return string in form '55%'
+	 */
+	function getOverlayWidth(video) {
+		var adWidth = video.container.closest('.wikia-ad').offsetWidth,
```
Comment (bognix) [inline extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js]: you already have container - it's passed as `params.container` in `add(..., params)` method.

File: extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+			videoContainerWidthString = video.container.style.width,
+			videoContainerWidth = parseInt(videoContainerWidthString.substr(0, videoContainerWidthString.length -2));
```
Comment (bognix) [inline extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js]: you already have this width - it's in `params.hideWhenPlaying.offsetWidth`.

File: skins/oasis/css/core/ads.scss [hunk 6]
```diff
@@ -177,6 +177,18 @@ $uap-video-animation-duration: 0.4s;
 	}
 }
 
+#BOTTOM_LEADERBOARD.bfab-template {
```
Comment (bognix) [inline skins/oasis/css/core/ads.scss]: you don't need to style with id

File: extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+	 *
+	 * offsetWidth won't work in case video container is hidden.
+	 * @param video
+	 * @return string in form '55%'
+	 */
+	function getOverlayWidth(video, params) {
+		var adWidth = params.container.offsetWidth,
+			videoContainerWidthString = video.container.style.width,
+			videoContainerWidth = parseInt(videoContainerWidthString.substr(0, videoContainerWidthString.length -2));
+
```
Comment (bognix) [inline extensions/wikia/AdEngine/js/video/player/ui/replayOverlay.js:50]: use `params.hideWhenPlaying.offsetWidth`

**Review by bognix — APPROVED** [review]
LGTM

**Comment by bognix** [comment]
Travis failure not related - merging.

**Pull request merged**